### ISSUE TITLE
Wrapper configuration not reloaded in a resubmission

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1124]
In GitLab by @ainagaya on Aug 28, 2023, 11:10

Hello @dbeltrankyl and @kinow, CC @mcastril 

### Autosubmit Version
```
2023-08-01 13:46:32,005 Autosubmit is running with 4.0.0b0
2023-08-01 13:46:32,005 Preparing run
```
### Expid affected(If applicable)

a07p (the user is iwagner)

### Which task has issues? Where is the log(If applicable)

* **Full_name**: `a07p_ASThread_1690543887613_3840_24`

* **Log_Path:** `/appl/AS/AUTOSUBMIT_DATA/a07p/tmp`

### Summary

We had an issue with the custom directives. Before the version was updated in the VM, the experiment started and failed. It was resubmitted (again with the wrong custom directives), I fixed the `main.yml` commenting the CUSTOM_DIRECTIVES keys, the submitted job was killed by LUMI admins, and then AS resubmitted the job (8/08) again with the wrong header. Now, with the updated version, the header is fine but it includes the CUSTOM_DIRECTIVES again (even if they are commented in the config files). **So we think that Autosubmit is not recalculating the configuration when resubmitting a wrapper after a failure.**

### Steps to reproduce

Put a wrapper in an experiment, change the configuration in the fly and make the wrapper fail. See if the configuration changed accordingly. 

### What is the current bug behavior?

Autosubmit is not recalculating the configuration when resubmitting a wrapper after a failure.

### What is the expected correct behavior?

Autosubmit recalculates the configuration when resubmitting a wrapper after a failure.

### Relevant logs and/or screenshots(if applicable)

Cancellation of the job by Lumi admins:

```
348 of 364 jobs remaining (11:21)
2023-08-08 11:21:35,202 Sleep: 10
2023-08-08 11:21:35,202 Number of retrials: 0
2023-08-08 11:21:35,205 Checking jobs for platform=LUMI
2023-08-08 11:21:35,206 WRAPPER CHECK TIME = 10
2023-08-08 11:21:35,206 Checking Wrapper 4324316
2023-08-08 11:21:37,915 Successful check job command: sacct -n -X --jobs 4324316 -o "State"
2023-08-08 11:21:37,915 Wrapper job a07p_ASThread_16911913926716_3840_24 changed from QUEUING to status FAILED
2023-08-08 11:21:37,915 Checking inner jobs status
2023-08-08 11:21:37,920 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_13_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,921 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_14_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,922 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_15_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,923 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_16_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,924 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_17_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,925 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_18_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,926 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_19_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,927 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_20_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,928 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_21_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,929 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_22_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,930 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_23_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,930 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_24_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,931 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_25_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,932 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_26_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,933 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_27_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,934 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_28_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,935 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_29_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,935 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_30_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,936 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_31_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,937 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_32_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,938 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_33_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,939 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_34_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,939 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_35_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
2023-08-08 11:21:37,940 /scratch/project_465000454/iwagner/a07p/LOG_a07p/a07p_19900101_fc0_36_SIM_FAILED File does not exist.. waiting 5s for a new retry (retries left: 3)
[ERROR] 2023-08-08 11:21:37,940 Cancelling job with id 4324316[eCode=6009]
2023-08-08 11:21:38,638 No jobs to check for platform LUMI
2023-08-08 11:21:38,639 Updating FAILED jobs
2023-08-08 11:21:38,639 Updating WAITING jobs
2023-08-08 11:21:38,639 Setting job: a07p_19900101_fc0_13_SIM status to: READY (all parents completed)...
2023-08-08 11:21:38,640 Update finished
2023-08-08 11:21:38,641 Saving JobList: /appl/AS/AUTOSUBMIT_DATA/a07p/pkl/job_list_a07p.pkl
2023-08-08 11:21:38,642 Job list saved
2023-08-08 11:21:38,645 Number of jobs available: 20
2023-08-08 11:21:38,645 Number of jobs ready: 1
2023-08-08 11:21:38,647 Jobs ready for LUMI: 1
2023-08-08 11:21:38,652 Saving JobList: /appl/AS/AUTOSUBMIT_DATA/a07p/pkl/job_list_a07p.pkl
2023-08-08 11:21:38,653 Job list saved
2023-08-08 11:21:38,654 
Jobs ready for LUMI: 1
2023-08-08 11:21:41,722 Creating Scripts
2023-08-08 11:21:41,799 Sending Files
2023-08-08 11:21:41,799 Check remote dir
2023-08-08 11:21:42,521 Compressing multiple_files
2023-08-08 11:21:42,537 Uncompress - send_command
2023-08-08 11:21:43,285 Send_file: common_script
2023-08-08 11:21:43,291 Submitting
2023-08-08 11:21:44,040 Saving JobList: /appl/AS/AUTOSUBMIT_DATA/a07p/pkl/job_list_a07p.pkl
2023-08-08 11:21:44,042 Job list saved
2023-08-08 11:21:44,043 Saving JobList: /appl/AS/AUTOSUBMIT_DATA/a07p/pkl/job_list_a07p.pkl
2023-08-08 11:21:44,044 Job list saved
2023-08-08 11:21:45,023 Saving JobList: /appl/AS/AUTOSUBMIT_DATA/a07p/pkl/job_list_a07p.pkl
2023-08-08 11:21:45,024 Job list saved
2023-08-08 11:21:45,028 Updating FAILED jobs
2023-08-08 11:21:45,028 Updating WAITING jobs
2023-08-08 11:21:45,029 Update finished
2023-08-08 11:21:45,031 Saving JobList: /appl/AS/AUTOSUBMIT_DATA/a07p/pkl/job_list_a07p.pkl
2023-08-08 11:21:45,032 Job list saved
2023-08-08 11:21:45,036 Backing up jobs_data...
2023-08-08 11:21:45,048 Jobs_data database backup completed.
2023-08-08 11:21:55,059 Reloading parameters...
2023-08-08 11:21:55,059 Loading parameters...
2023-08-08 11:21:55,060 Parameters load.
2023-08-08 11:21:55,061 

348 of 364 jobs remaining (11:21)
2023-08-08 11:21:55,061 Sleep: 10
2023-08-08 11:21:55,061 Number of retrials: 0
2023-08-08 11:21:55,067 Checking jobs for platform=LUMI
2023-08-08 11:21:55,068 WRAPPER CHECK TIME = 10
2023-08-08 11:21:55,068 Checking Wrapper 4343071
2023-08-08 11:21:57,803 Successful check job command: sacct -n -X --jobs 4343071 -o "State"
2023-08-08 11:21:57,803 Wrapper job a07p_ASThread_16914828989330_3840_24 changed from SUBMITTED to status QUEUING
2023-08-08 11:21:57,803 Checking inner jobs status
2023-08-08 11:21:58,510 Job a07p_ASThread_16914828989330_3840_24 is QUEUING (Priority)
2023-08-08 11:21:58,511 No jobs to check for platform LUMI
2023-08-08 11:21:58,511 Updating FAILED jobs
```

### Any other relevant information(if applicable)

CC @Lerriola FYI also

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 28, 2023, 11:16

We verified that the `main.yml`, containing the CUSTOM_DIRECTIVES, had been modified the day before.


`-rw-rw----. 1 iwagner autosubmit_users 956 Aug  7 09:06 main.yml`

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 18, 2024, 08:40

Related to #1238 , a similar issue in v4.4 will fix some issues regarding jobs inside wrappers not having their parameters updated

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 11, 2024, 11:04

Should be fixed already, at least in 4.1.7, if not please open a new issue and tag this one
